framework/src/base/NonlinearEigenSystem.C

```diff
@@ -135,4 +145,14 @@ NonlinearEigenSystem::getNthConvergedEigenvalue(dof_id_type n)
   return _transient_sys.get_eigenpair(n);
 }
 
+void
+NonlinearEigenSystem::addKernelToExtraWarehouse(MooseSharedPointer<KernelBase> kernel, THREAD_ID tid)
```

Comment:
This routine handles the EigenKernels only. I say `ExtraWarehouse` means if we need to handle other kernels which can not be handled by `NonlinearSystemBase`.  `All kernels`, and "Time kernels" are handled in `NonlinearSystemBase`.  My motivation is to make the new implementation consistent with the existing MOOSE.